### PR TITLE
Move WAL overwrite under switch and add precaution checks

### PR DESCRIPTION
**Pull request by x4m**
@DustinChaloupka, @fdr plz review this PR, mainly design decisions. I'll add docs later.

Here we check WAL contents if WAL-G tries to overwrite WAL.

I've tried MD5==ETag thing, it does not work for gpg-encrypted files, because the file is different each time.

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -617,6 +610,42 @@ func UploadWALFile(tarUploader *TarUploader, dirArc string, pre *S3Prefix, verif
...
+		if err != nil {
+			log.Fatalf("FATAL: %+v\n", err)
+		}
+
+		if !bytes.Equal(archived.Bytes(), localBytes) {
+			log.Fatalf("FATAL: WAL file '%s' already archived, contents differ, unable to overwrite\n", dirArc)
+		} else {
+			log.Printf("WARNING: WAL file '%s' already archived, archived content equals\n", dirArc)
+		}
+	}
```
Comment (DustinChaloupka) [inline commands.go]: This function looks fine design-wise, the only thing that might potentially be a worry is how much slower this makes the uploads.

File: archive.go [hunk 2]
```diff
@@ -15,24 +15,29 @@ type Archive struct {
...
-				return false, nil
+				return false, headResult.ETag, nil
 			default:
-				return false, awsErr
+				return false, headResult.ETag, awsErr
 			}
 		}
 	}
-	return true, nil
+	return true, headResult.ETag, nil
```
[the remark this answers is not in the document]
Comment (x4m) [reply archive.go]: yeah, that's what left from MD5 experiments, will delete these lines.

File: commands.go (hunk 1 above)
Comment (x4m) [reply commands.go]: well, uploads are reversed: you download and compare. I didn't benchmark though, I think it is not a normal situation when you try to overwrite WAL, even if with the same data.

**Comment by x4m** [comment]
@Tinsane brought my attention to one more thing: we should not report error (with nonzero exit code) if overwrite is attempted from bgUploader, only if we encounter overwrite with main file.

**Pull request merged**